File: ExampleApp/app/src/main/java/com/motiv/example/UsersListAdapter.java
package com.motiv.example;

import android.view.LayoutInflater;
import android.view.View;
import androidx.recyclerview.widget.*;
import com.motiv.example.databinding.UserslistadapterBinding;
import com.squareup.picasso.Picasso;
import java.util.*;
import java.util.ArrayList;
import java.util.concurrent.*;

public class UsersListAdapter extends RecyclerView.Adapter<UsersListAdapter.AdapterViewHolder> {

    private OnItemClickListener onItemClickListener;
    private ArrayList<User> data = new ArrayList();

    public interface OnItemClickListener {

        public void onItemClick(int position, com.motiv.example.User item);
    }

    public class AdapterViewHolder extends RecyclerView.ViewHolder {

        private UserslistadapterBinding binding;

        public AdapterViewHolder(UserslistadapterBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }

    @Override
    public int getItemCount() {
        return data.size();
    }

    public User getElement(int position) {

        return data.get(position);
    }

    public void setData(java.util.List<com.motiv.example.User> newData) {
        data.clear();
        data.addAll(newData);
        notifyDataSetChanged();
    }

    @Override
    public AdapterViewHolder onCreateViewHolder(android.view.ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        UserslistadapterBinding binding = UserslistadapterBinding.inflate(inflater);
        return new AdapterViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(final AdapterViewHolder viewHolder, final int position) {
        viewHolder.itemView.setOnClickListener(
                new View.OnClickListener() {
                    @Override
                    public void onClick(View v) {

                        if (onItemClickListener != null)
                            onItemClickListener.onItemClick(position, data.get(position));
                    }
                });
        User user = data.get(position);
        viewHolder.binding.textview11.setText(user.getFirst_name());

        Picasso.with(viewHolder.itemView.getContext())
                .load(user.getLinks().getAvatar().getHref())
                .into(viewHolder.binding.imageview10);
        ;
    }

    public void setOnItemClickListener(
            com.motiv.example.UsersListAdapter.OnItemClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
    }
}

File: ExampleApp/app/src/main/java/com/motiv/example/UsersListAdapter.kt
 
package com.motiv.example
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.fragment.app.*
import androidx.recyclerview.widget.*
import com.bumptech.glide.Glide
import com.motiv.example.databinding.UserslistadapterBinding
import dagger.*
import dagger.android.*
import dagger.android.support.*
import javax.inject.*
import kotlin.collections.List
import kotlinx.android.synthetic.main.userslistadapter.view.*

public class UsersListAdapter : RecyclerView.Adapter<UsersListAdapter.AdapterViewHolder>() {

    private var onItemClickListener: OnItemClickListener? = null

    private var data: ArrayList<User> = ArrayList()

    public interface OnItemClickListener {

        public fun onItemClick(position: Int, item: com.motiv.example.User): Unit
    }

    public class AdapterViewHolder(val binding: UserslistadapterBinding) : RecyclerView.ViewHolder(binding.getRoot())

    override fun getItemCount(): Int {
        return data.size
    } fun getElement(position: Int): User {
        return data.get(position)
    } fun setData(newData: List<com.motiv.example.User>) {
        data.clear()
        data.addAll(newData)
        notifyDataSetChanged()
    } override fun onCreateViewHolder(parent: android.view.ViewGroup, viewType: Int): AdapterViewHolder {
        val inflater: LayoutInflater = LayoutInflater.from(parent.context)
        val binding: UserslistadapterBinding = UserslistadapterBinding.inflate(inflater)
        return AdapterViewHolder(binding)
    } override fun onBindViewHolder(viewHolder: AdapterViewHolder, position: Int) {
        viewHolder.itemView.setOnClickListener {
            if (onItemClickListener != null)
                onItemClickListener?.onItemClick(position, data.get(position))
        }
        val user: User = data.get(position)
        viewHolder.binding.textview11.setText(user.getFirst_name())
        Glide.with(viewHolder.itemView.getContext())
            .load(user.getLinks().getAvatar().getHref())
            .into(viewHolder.binding.imageview10)
    } fun setOnItemClickListener(onItemClickListener: com.motiv.example.UsersListAdapter.OnItemClickListener) {
        this.onItemClickListener = onItemClickListener
    }
}
